File: go_blockchain/two_proof_of_work/blc/block.go
package blc

import (
	"time"
)

//区块结构体
type Block struct {
	//1.区块高度
	Height int64
	//2.上一个区块的hash值
	PrevBlockHash []byte
	//3.交易数据
	Data []byte
	//4.时间戳
	Timestamp int64
	//5.hash值
	Hash []byte
	// 6. Nonce
	Nonce int64
}

//创建新的区块
func NewBlock(data string, height int64, prevBlockHash []byte) *Block {

	block := &Block{height, prevBlockHash, []byte(data), time.Now().Unix(), nil, 0}

	//调用工作量证明的方法并且返回有效的Hash和Nonce
	pow := NewProofOfWork(block)
	hash,nonce := pow.Run()

	block.Hash = hash[:]
	block.Nonce = nonce

	return block
}


//生成创世区块
func CreateGenesisBlock(data string) *Block {
	return NewBlock(data, 1, []byte{0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0})
}

File: java_blockchain/blockchain/src/main/java/com/whoiszxl/blc/model/po/Block.java
package com.whoiszxl.blc.model.po;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Builder
public class Block {

	/** 区块高度 */
	private Long height;
	
	/** 上一个区块的hash值  */
	private String prevBlockHash;
	
	/** 交易数据  */
	private String data;

	/** 时间戳 */
	private Long timestamp;
	
	/** hash值 */
	private String hash;
	
	/** 工作量校驗值 */
	private Long nonce;
}
